Incorrect test status in QuarkusTestAfterTestExecutionCallback and QuarkusTestAfterEachCallback for failing test

### Describe the bug

When an exception is raised in a test method of a `@QuarkusTest` annotated class, the `QuarkusTestAfterTestExecutionCallback` and `QuarkusTestAfterEachCallback` callbacks receive an incorrect `TestStatus` that does not report the error.


### Expected behavior

When an exception is raised in a test, `QuarkusTestMethodContext.getStatus()` in `QuarkusTestAfterTestExecutionCallback` and `QuarkusTestAfterEachCallback` callbacks should report the exception.

### Actual behavior

The callbacks are invoked, but `QuarkusTestMethodContext.getStatus().isTestFailed()` always return `false`.

### How to Reproduce?

Create a class implementing the callbacks and register it in `META-INF/services/io.quarkus.test.junit.callback.QuarkusTestAfterEachCallback` and `META-INF/services/io.quarkus.test.junit.callback.QuarkusTestAfterTestExecutionCallback`.

```java
public class CustomCallbacks implements QuarkusTestAfterTestExecutionCallback, QuarkusTestAfterEachCallback {
    
    @Override
    public void afterEach(QuarkusTestMethodContext context) {
        System.out.println("afterEach :: isTestFailed = " + context.getTestStatus().isTestFailed());
    }

    @Override
    public void afterTestExecution(QuarkusTestMethodContext context) {
        System.out.println("afterTestExecution :: isTestFailed = " + context.getTestStatus().isTestFailed());
    }
}
```

Then run a simple test that throws an exception and verify that `context.getTestStatus().isTestFailed()` is false

```java
@QuarkusTest
public class SimpleTest {

    @Test
    public void testCallbacks() {
        throw new RuntimeException("BOOM!");
    }
}
```

```
2023-05-27 16:55:35,629 INFO  [io.quarkus] (main) getting-started 1.0.0-SNAPSHOT on JVM (powered by Quarkus 3.0.4.Final) started in 1.305s. Listening on: http://localhost:8081
2023-05-27 16:55:35,644 INFO  [io.quarkus] (main) Profile test activated. 
2023-05-27 16:55:35,644 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy-reactive, smallrye-context-propagation, vertx]
afterTestExecution :: isTestFailed = false
afterEach :: isTestFailed = false

java.lang.RuntimeException: BOOM!

	at com.example.application.SimpleTest.testCallbacks(SimpleTest.java:14)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:999)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:813)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:156)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:147)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:86)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(InterceptingExecutableInvoker.java:103)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:86)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:217)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:213)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:138)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.8

### Additional information

_No response_

Here's a zip based on a project created with the quarkus maven plugin

[quarkus-test-callback.zip](https://github.com/quarkusio/quarkus/files/11591829/quarkus-test-callback.zip)

If you run `mvn test`, you can see in the console the reported output,

```
[INFO] Running com.example.application.SimpleTest
2023-05-29 14:44:53,249 INFO  [io.quarkus] (main) getting-started 1.0.0-SNAPSHOT on JVM (powered by Quarkus 3.0.4.Final) started in 1.318s. Listening on: http://localhost:8081
2023-05-29 14:44:53,267 INFO  [io.quarkus] (main) Profile test activated. 
2023-05-29 14:44:53,267 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy-reactive, smallrye-context-propagation, vertx]
afterTestExecution :: isTestFailed = false
afterEach :: isTestFailed = false
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 1.978 s <<< FAILURE! - in com.example.application.SimpleTest
[ERROR] com.example.application.SimpleTest.testCallbacks  Time elapsed: 0.187 s  <<< ERROR!
java.lang.RuntimeException: BOOM!
	at com.example.application.SimpleTest.testCallbacks(SimpleTest.java:14)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
```


In particular, the following lines, that make me think there may be an issue with the parameter passed to the callbacks
```
afterTestExecution :: isTestFailed = false
afterEach :: isTestFailed = false
```

If the test fails with an exception, I would expect `context.getTestStatus().isTestFailed()` to return `true`

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/QuarkusTestCallbacksTestCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/QuarkusTestCallbacksTestCase.java
@@ -114,6 +114,12 @@ public class QuarkusTestCallbacksTestCase {
         assertTrue(TestContextCheckerBeforeEachCallback.CONTEXT.getTestStatus().isTestFailed());
     }
 
+    @Test
+    @Order(6)
+    public void testCallbackContextIsNotFailedAgain() {
+        assertTrue(TestContextCheckerBeforeEachCallback.CONTEXT.getTestStatus().isTestFailed());
+    }
+
     @Target({ METHOD })
     @Retention(RUNTIME)
     public @interface TestAnnotation {

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/AbstractQuarkusTestWithContextExtension.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/AbstractQuarkusTestWithContextExtension.java
@@ -4,12 +4,20 @@ import java.io.IOException;
 
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.api.extension.LifecycleMethodExecutionExceptionHandler;
+import org.junit.jupiter.api.extension.TestExecutionExceptionHandler;
 import org.junit.jupiter.api.extension.TestWatcher;
 
 public abstract class AbstractQuarkusTestWithContextExtension extends AbstractTestWithCallbacksExtension
-        implements LifecycleMethodExecutionExceptionHandler, TestWatcher {
+        implements TestExecutionExceptionHandler, LifecycleMethodExecutionExceptionHandler, TestWatcher {
 
     public static final String IO_QUARKUS_TESTING_TYPE = "io.quarkus.testing.type";
+
+    @Override
+    public void handleTestExecutionException(ExtensionContext context, Throwable throwable) throws Throwable {
+        markTestAsFailed(context, throwable);
+
+        throw throwable;
+    }
 
     @Override
     public void handleAfterAllMethodExecutionException(ExtensionContext context, Throwable throwable) throws Throwable {

